Unable to install

I get a permission denied error when trying to install.

```
firas@dev:~/test$ npm i pcre2

> pcre2@0.0.4-a install /home/firas/test/node_modules/pcre2
> node-gyp rebuild

make: Entering directory '/home/firas/test/node_modules/pcre2/build'
  ACTION Building PCRE2 library... .
/bin/sh: 1: eval: ./compile: Permission denied
pcre2.target.mk:13: recipe for target '.' failed
make: *** [.] Error 126
make: Leaving directory '/home/firas/test/node_modules/pcre2/build'
gyp ERR! build error
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:258:23)
gyp ERR! stack     at emitTwo (events.js:135:13)
gyp ERR! stack     at ChildProcess.emit (events.js:224:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:209:12)
gyp ERR! System Linux 3.16.0-4-amd64
gyp ERR! command "/usr/local/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /home/firas/test/node_modules/pcre2
gyp ERR! node -v v9.0.0
gyp ERR! node-gyp -v v3.6.2
gyp ERR! not ok
```

--- a/compile
+++ b/compile
@@ -4,6 +4,6 @@ mkdir -p build_pcre2/
 
 cd deps/pcre2/
 
-sh ./configure --with-parens-nest-limit=50000 --with-match-limit=500000000 --disable-stack-for-recursion --enable-static --disable-shared --with-pic --prefix=$PWD/../../build_pcre2/
+sh ./configure --with-parens-nest-limit=50000 --with-match-limit=500000000 --enable-static --disable-shared --with-pic --prefix=$PWD/../../build_pcre2/
 make
 make install

--- a/deps/jpcre2/src/jpcre2.hpp
+++ b/deps/jpcre2/src/jpcre2.hpp
@@ -3876,6 +3876,16 @@ struct select{
             return *pat_str_ptr; 
         }
         
+        ///Get number of captures from compiled code.
+        ///@return New line option value or 0.
+        Uint getNumCaptures() {
+            if(!code) return 0;
+            Uint numCaptures = 0;
+            int ret = Pcre2Func<sizeof( Char_T ) * CHAR_BIT>::pattern_info(code, PCRE2_INFO_CAPTURECOUNT, &numCaptures);
+            if(ret < 0) error_number = ret;
+            return numCaptures;
+        }
+
         /// Get pointer to pattern string
         ///@return Pointer to constant pattern string
         String const * getPatternPointer() const  { 

--- a/deps/jpcre2/src/test.cpp
+++ b/deps/jpcre2/src/test.cpp
@@ -85,6 +85,7 @@ int main(){
     re.getErrorNumber(); \
     re.getErrorOffset(); \
     re.getPattern(); \
+    re.getNumCaptures(); \
     re.getPatternPointer(); \
     re.getPcre2Option(); \
     re.getJpcre2Option(); \

--- a/deps/pcre2/.gitignore
+++ b/deps/pcre2/.gitignore
@@ -3,7 +3,6 @@ config.status
 config.log
 libtool
 
-src/*.c
 src/*.o
 src/*.lo
 src/config.h

--- a/src/PCRE2Wrapper.cc
+++ b/src/PCRE2Wrapper.cc
@@ -34,7 +34,7 @@ void PCRE2Wrapper::Init(std::string name, v8::Local<v8::Object> exports) {
 	Nan::SetAccessor(proto, Nan::New("sticky").ToLocalChecked(), PCRE2Wrapper::PropertyGetter);
 
 	constructor.Reset(Nan::GetFunction(tpl).ToLocalChecked());
-	exports->Set(Nan::New(name).ToLocalChecked(), Nan::GetFunction(tpl).ToLocalChecked());
+	exports->Set(Nan::GetCurrentContext(), Nan::New(name).ToLocalChecked(), Nan::GetFunction(tpl).ToLocalChecked());
 }
 
 void PCRE2Wrapper::New(const Nan::FunctionCallbackInfo<v8::Value>& info) {
@@ -148,22 +148,22 @@ void PCRE2Wrapper::Exec(const Nan::FunctionCallbackInfo<v8::Value>& info) {
 		std::string whole_match = obj->vec_num[0][0];
 		
 		for ( size_t i=0; i<obj->vec_num[0].size(); i++ ) {
-			result->Set(i, Nan::New(obj->vec_num[0][i]).ToLocalChecked());
+			result->Set(Nan::GetCurrentContext(), i, Nan::New(obj->vec_num[0][i]).ToLocalChecked());
 		}
 				
 		v8::Local<v8::Object> named = Nan::New<v8::Object>();
-		result->Set(Nan::New("groups").ToLocalChecked(), named);
-		result->Set(Nan::New("named").ToLocalChecked(), named);
-	
+		result->Set(Nan::GetCurrentContext(), Nan::New("groups").ToLocalChecked(), named);
+		result->Set(Nan::GetCurrentContext(), Nan::New("named").ToLocalChecked(), named);
+
 		for ( auto const& ent : obj->vec_nas[0] ) {
-			named->Set(Nan::New(ent.first).ToLocalChecked(), Nan::New(ent.second).ToLocalChecked());
+			named->Set(Nan::GetCurrentContext(), Nan::New(ent.first).ToLocalChecked(), Nan::New(ent.second).ToLocalChecked());
 		}
 		
 		int match_offset = finish_offset - whole_match.length();
-		
-		result->Set(Nan::New("index").ToLocalChecked(), Nan::New((int32_t)match_offset));
-		result->Set(Nan::New("input").ToLocalChecked(), Nan::New(subject).ToLocalChecked());
-		
+
+		result->Set(Nan::GetCurrentContext(), Nan::New("index").ToLocalChecked(), Nan::New((int32_t)match_offset));
+		result->Set(Nan::GetCurrentContext(), Nan::New("input").ToLocalChecked(), Nan::New(subject).ToLocalChecked());
+
 		info.GetReturnValue().Set(result);
 	}
 }
@@ -202,7 +202,7 @@ void PCRE2Wrapper::Match(const Nan::FunctionCallbackInfo<v8::Value>& info) {
 		v8::Local<v8::Array> result = Nan::New<v8::Array>(result_count);
 		
 		for ( size_t i=0; i<obj->vec_num.size(); ++i ) {
-			result->Set(i, Nan::New(obj->vec_num[i][0]).ToLocalChecked());
+			result->Set(Nan::GetCurrentContext(), i, Nan::New(obj->vec_num[i][0]).ToLocalChecked());
 		}
 		
 		info.GetReturnValue().Set(result);
@@ -268,7 +268,7 @@ void PCRE2Wrapper::Replace(const Nan::FunctionCallbackInfo<v8::Value>& info) {
 				argVector[argCount-1] = named; //named
 				
 				for ( auto const& ent : ng ) {
-					named->Set(Nan::New(ent.first).ToLocalChecked(), Nan::New(ent.second).ToLocalChecked());
+					named->Set(Nan::GetCurrentContext(), Nan::New(ent.first).ToLocalChecked(), Nan::New(ent.second).ToLocalChecked());
 				}
 
 				v8::Local<v8::Value> returned = Nan::Callback(callback).Call(argCount, argVector);

--- a/src/PCRE2Wrapper.cc
+++ b/src/PCRE2Wrapper.cc
@@ -254,11 +254,15 @@ void PCRE2Wrapper::Replace(const Nan::FunctionCallbackInfo<v8::Value>& info) {
 				const jpcre2::VecOff* vec_soff = me.getMatchStartOffsetVector();
 				size_t match_offset = (*vec_soff)[pos];
 				
-				const unsigned argCount = 3 + cg.size();
+				uint32_t numCaptures = obj->re.getNumCaptures();
+				const unsigned argCount = 3 + numCaptures + 1;
 				v8::Local<v8::Value> *argVector = new v8::Local<v8::Value>[argCount];
 
 				for ( size_t i=0; i<cg.size(); i++ ) {
 					argVector[i] = Nan::New(cg[i]).ToLocalChecked(); //match, p1, p2, ... , pn
+				}
+				for ( size_t i=cg.size(); i<numCaptures+1; i++ ) {
+					argVector[i] = Nan::Null(); // pad with null
 				}
 
 				argVector[argCount-3] = Nan::New((uint32_t)match_offset); //offset

--- a/src/PCRE2Wrapper.h
+++ b/src/PCRE2Wrapper.h
@@ -1,6 +1,8 @@
 #ifndef PCRE2Wrapper_H
 #define PCRE2Wrapper_H
 
+#include <climits> // CHAR_BIT
+
 #include <nan.h>
 #include <jpcre2.hpp>
 

